languages/go/training.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import (
+  "fmt"
+  "reflect"
+)
+
+type SampleType struct {
+  id string
+  name string
+  description string
+}
+
+type SliceMetrics struct {
+  t reflect.Type
+  length int
+  capacity int
+  value string
+}
+
+func main() {
+  createSliceTests()
+  createMultiTypeSliceTests()
+  reverseStringTests()
+  getFibRecursiveTests()
+  getFibIterativeTests()
+
+  fmt.Println("All Tests Passed!")
+}
+
+func createSlice[T any](values ...T) []T {
+  slice := []T{}
+  
+  for _, value := range values {
+    slice = append(slice, value)
+  }
+  
+  return slice
+}
+
+func createSliceTests() {
+  if !reflect.DeepEqual(createSlice[any](), []any{}) {
+    panic("createSlice created an incorrect value for empty-any list")
+  }
+  if !reflect.DeepEqual(createSlice[int](), []int{}) {
+    panic("createSlice created an incorrect value for empty-int list")
+  }
+  if !reflect.DeepEqual(createSlice[int32](), []int32{}) {
+    panic("createSlice created an incorrect value for empty-char list")
+  }
+  if !reflect.DeepEqual(createSlice[string](), []string{}) {
+    panic("createSlice created an incorrect value for empty-string list")
+  }
+  if !reflect.DeepEqual(createSlice[any](1.0), []any{1.0}) {
+    panic("createSlice created an incorrect value for single-any list")
+  }
+  if !reflect.DeepEqual(createSlice(1), []int{1}) {
+    panic("createSlice created an incorrect value for single-int list")
+  }
+  if !reflect.DeepEqual(createSlice('a'), []int32{'a'}) {
+    panic("createSlice created an incorrect value for single-char list")
+  }
+  if !reflect.DeepEqual(createSlice("test"), []string{"test"}) {
+    panic("createSlice created an incorrect value for single-string list")
+  }
+  if !reflect.DeepEqual(createSlice[any](1,1.0,'a',"test",nil), []any{1,1.0,'a',"test",nil}) {
+    panic("createSlice created an incorrect value for multi-any list")
+  }
+  if !reflect.DeepEqual(createSlice(1,2,3,4,5), []int{1,2,3,4,5}) {
+    panic("createSlice created an incorrect value for multi-int list")
+  }
+  if !reflect.DeepEqual(createSlice('a','b','c','d','e'), []int32{'a','b','c','d','e'}) {
+    panic("createSlice created an incorrect value for multi-char list")
+  }
+  if !reflect.DeepEqual(createSlice("test","abc","123","...",""), []string{"test","abc","123","...",""}) {
+    panic("createSlice created an incorrect value for multi-string list")
+  }
+}
+
+func createMultiTypeSlice(values ...any) []any {
+  slice := []any{}
+  
+  for _, value := range values {
+    slice = append(slice, value)
+  }
+  
+  return slice
+}
+
+func createMultiTypeSliceTests() {
+  if !reflect.DeepEqual(createMultiTypeSlice(), []any{}) {
+    panic("createSlice created an incorrect value for empty list")
+  }
+  if !reflect.DeepEqual(createMultiTypeSlice(1,1.0,'a',"test"), []any{1,1.0,'a',"test"}) {
+    panic("createSlice created an incorrect value for multi-type list")
```

Comment:
Let's update these panic messages to reflect the correct underlying function names, e.g. `createMultiTypeSlice`